src/main/java/hudson/plugins/ec2/EC2Cloud.java

```diff
@@ -1134,7 +1174,7 @@ public FormValidation doCheckSshKeysCredentialsId(@AncestorInPath ItemGroup cont
                 return FormValidation.ok();
             }
             if (value == null || value.isEmpty()){
-                return FormValidation.error("No ssh credentials selected");
+                return FormValidation.ok("No SSH credentials selected, so SSH key management will be handled dynamically by Jenkins");
```

Comment:
It would be helpful in this case in `doTestConnection` to actually try generating and then deleting a keypair, to validate that the IAM principal has the corresponding permissions. Otherwise someone who originally configured the plugin with a static keypair by selecting the minimal EC2 permissions in some policy might try switching to dynamic mode and only see it fail when trying to run builds.